FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -212,7 +227,126 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
             }
         }
 
+        private void UpdateSubmittingJobStatus(FhirToDataLakeProcessingJobInputData input, long newJobId)
+        {
+            _jobStatus.CreatedJobCount++;
+            if (_inputData.JobVersion == JobVersion.V1 || _inputData.JobVersion == JobVersion.V2)
+            {
+                _jobStatus.RunningJobIds.Add(newJobId);
+            }
+
+            _jobStatus.SequenceIdToJobIdMapForRunningJobs[input.ProcessingJobSequenceId] = newJobId;
+
+            if (input.SplitParameters != null)
+            {
+                foreach (var param in input.SplitParameters)
+                {
+                    _jobStatus.SubmittedResourceTimestamps[param.Key] = param.Value.DataEndTime;
+                }
+            }
+
+            _jobStatus.SubmittingProcessingJob = null;
+        }
+
+        private FhirToDataLakeOrchestratorJobStatus InitializeJobStatusFromResult()
+        {
+            var resultContent = JsonConvert.SerializeObject(_result);
```

Comment:
Maybe we can directly copy members instead of deserialization? Just in case of definitions changes in the source/target data structure

Reply:
Updated it , thanks.